cmd/bnsd/scenarios/escrow_test.go

```diff
@@ -0,0 +1,69 @@
+package scenarios
+
+import (
+	"testing"
+	"time"
+
+	"github.com/iov-one/weave/cmd/bnsd/app"
+	"github.com/iov-one/weave/cmd/bnsd/client"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/x/escrow"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestQueryEscrowExists(t *testing.T) {
+	walletResp, err := bnsClient.GetWallet(escrowContract.Address())
+	// then
+	require.NoError(t, err)
+	require.NotNil(t, walletResp)
+	require.Len(t, walletResp.Wallet.Coins, 1)
+	assert.True(t, walletResp.Wallet.Coins[0].Whole > 0)
+}
+
+func TestEscrowRelease(t *testing.T) {
+	// query distribution accounts start balance
+	walletResp, err := bnsClient.GetWallet(distrContractAddr)
+	require.NoError(t, err)
+	startBalance := coin.Coin{Ticker: "IOV"}
+	if walletResp != nil {
+		startBalance = *walletResp.Wallet.Coins[0]
+	}
+	aNonce := client.NewNonce(bnsClient, alice.PublicKey().Address())
+	// when releasing 1 IOV by the arbiter
+	_, _, escrowID, _ := escrowContract.Parse()
+
+	addValidatorTX := &app.Tx{
+		Sum: &app.Tx_ReleaseEscrowMsg{
+			&escrow.ReleaseEscrowMsg{
+				EscrowId: escrowID,
+				Amount:   []*coin.Coin{{Ticker: "IOV", Whole: 1}},
+			},
+		},
+	}
+	_, _, contractID, _ := multiSigContract.Parse()
+	addValidatorTX.Multisig = [][]byte{contractID}
+
+	seq, err := aNonce.Next()
+	require.NoError(t, err)
+	require.NoError(t, client.SignTx(addValidatorTX, alice, chainID, seq))
+	resp := bnsClient.BroadcastTx(addValidatorTX)
+
+	// then
+	require.NoError(t, resp.IsError())
+
+	// and check it was added to the distr account
+	for i := 0; 1 < 10; i++ {
+		walletResp, err = bnsClient.GetWallet(distrContractAddr)
+		require.NoError(t, err)
+		if walletResp.Height > resp.Response.Height {
+			break
+		}
+		time.Sleep(200 * time.Millisecond)
+	}
+	require.NotNil(t, walletResp)
+	require.True(t, walletResp.Height > resp.Response.Height)
+	require.True(t, len(walletResp.Wallet.Coins) == 1)
+	// new balance should be higher
+	assert.False(t, startBalance.IsGTE(*walletResp.Wallet.Coins[0]), "%s not > %s", *walletResp.Wallet.Coins[0], startBalance)
```

Comment:
I mean, I guess we could assert there is a difference of exactly 1 token (as that is what is released) to show the whole contract is not released, but this is a general sanity check as well

Reply:
We have good coverage of this functionality in unit tests already. In scenarios I kept the asserts very generous as the target system may be used by concurrent users. This more a smoke test level of sanity checks